tests/upgrade_path/upgrade_helpers.py

```diff
@@ -114,7 +114,7 @@ def install_sonic(duthost, image_url, tbinfo):
             # Temporarily change the default route to mgmt-gateway address. This is done so that
             # DUT can download an image from a remote host over the mgmt network.
             logger.info("Add default mgmt-gateway-route to the device via {}".format(mg_gwaddr))
-            duthost.shell("ip route add default via {}".format(mg_gwaddr), module_ignore_errors=True)
+            duthost.shell("ip route add default via {} metric 1".format(mg_gwaddr), module_ignore_errors=True)
```

Comment:
How about using `replace` instead of `add`?
`ip route replace default via {}.format(mg_gwaddr)`

Reply:
If that will replace the default route then that is not what we want. After this step, when the new image is downloaded, we remove the newly installed route, and maintain the original default route (installed by BGP).
```
admin@vlab-01:~$ show ip route | more                              
Codes: K - kernel route, C - connected, S - static, R - RIP,
       O - OSPF, I - IS-IS, B - BGP, P - PIM, A - Babel,
       > - selected route, * - FIB route

K>* 0.0.0.0/0 via 10.250.0.1, eth0
B   0.0.0.0/0 [20/0] via 10.0.0.57, PortChannel101, src 10.1.0.32, 00:02:54
                     via 10.0.0.59, PortChannel102, src 10.1.0.32, 00:02:54
                     via 10.0.0.61, PortChannel103, src 10.1.0.32, 00:02:54
                     via 10.0.0.63, PortChannel104, src 10.1.0.32, 00:02:54
```

Deletion of this temporary route happens here:
https://github.com/Azure/sonic-mgmt/blob/8ce88c76c68a365e71911539bf426d83557af71a/tests/upgrade_path/upgrade_helpers.py#L141

Reply:
What's the metric value of the learnt default route (from BGP)? I saw it's 0 on my vtestbed.
`admin@vlab-01:~$ show ip route 0.0.0.0
Routing entry for 0.0.0.0/0
  Known via "bgp", distance 20, metric 0, best
  Last update 02w0d20h ago
  * 10.0.0.59, via PortChannel0002, weight 1
  * 10.0.0.61, via PortChannel0003, weight 1
  * 10.0.0.63, via PortChannel0004, weight 1

Routing entry for 0.0.0.0/0
  Known via "static", distance 200, metric 0
  Last update 03w1d00h ago
    10.250.0.1, via eth0, weight 1
`
Which route will be selected if the newly added route metric is `1`?

Reply:
> Which route will be selected if the newly added route metric is 1?

The route selected was still the one that is newly added (even if metric 1 is greater than already installed (metric 0).

This was a good catch, Bing. I realized that the reason the new installed route was preferred was not the lesser metric value.  I tried with much larger metric (100), and it still favored the new large-metric route.

If I didn't give a metric value, then the new route was failing to be installed for 201811 image where a default route via eth0 already exists as a static route, but not a preferred one.
This was not the case for 202012 image, where new route was installed even without any metric value - as there was no static route via eth0 in 202012.

Ultimately your first suggestion worked - replace instead of add. I verified that that replace will not delete old installed routes:

For 202012:
```
admin@vlab-01:~$ show ip route 0.0.0.0
Routing entry for 0.0.0.0/0
  Known via "bgp", distance 20, metric 0, best
  Last update 00:18:18 ago
  * 10.0.0.57, via PortChannel101, weight 1
  * 10.0.0.59, via PortChannel102, weight 1
  * 10.0.0.61, via PortChannel103, weight 1
  * 10.0.0.63, via PortChannel104, weight 1
admin@vlab-01:~$
admin@vlab-01:~$ sudo ip route replace default via 10.250.0.1 
admin@vlab-01:~$ 
admin@vlab-01:~$ show ip route 0.0.0.0
Routing entry for 0.0.0.0/0
  Known via "kernel", distance 0, metric 0, best
  Last update 00:00:02 ago
  * 10.250.0.1, via eth0
Routing entry for 0.0.0.0/0
  Known via "bgp", distance 20, metric 0
  Last update 00:18:24 ago
    10.0.0.57, via PortChannel101, weight 1
    10.0.0.59, via PortChannel102, weight 1
    10.0.0.61, via PortChannel103, weight 1
    10.0.0.63, via PortChannel104, weight 1
admin@vlab-01:~$ 
```

For 201811:
```
admin@vlab-01:~$ show ip route 0.0.0.0
Routing entry for 0.0.0.0/0
  Known via "bgp", distance 20, metric 0, best
  Last update 00:00:12 ago
  * 10.0.0.59, via PortChannel102, src 10.1.0.32
  * 10.0.0.57, via PortChannel101, src 10.1.0.32
  * 10.0.0.61, via PortChannel103, src 10.1.0.32
  * 10.0.0.63, via PortChannel104, src 10.1.0.32

Routing entry for 0.0.0.0/0
  Known via "static", distance 200, metric 0
    10.250.0.1, via eth0

admin@vlab-01:~$ sudo ip route replace default via 10.250.0.1 
admin@vlab-01:~$ 
admin@vlab-01:~$ show ip route 0.0.0.0
Routing entry for 0.0.0.0/0
  Known via "kernel", distance 0, metric 0, best
  * 10.250.0.1, via eth0

Routing entry for 0.0.0.0/0
  Known via "bgp", distance 20, metric 0
  Last update 00:00:34 ago
    10.0.0.59, via PortChannel102, src 10.1.0.32
    10.0.0.57, via PortChannel101, src 10.1.0.32
    10.0.0.61, via PortChannel103, src 10.1.0.32
    10.0.0.63, via PortChannel104, src 10.1.0.32

Routing entry for 0.0.0.0/0
  Known via "static", distance 200, metric 0
    10.250.0.1, via eth0
admin@vlab-01:~$ 
```